doc/source/serve/production-guide/fault-tolerance.md

```diff
@@ -0,0 +1,304 @@
+(serve-e2e-ft)=
+# Adding End-to-End Fault Tolerance
+
+Serve handles component failures _within_ a Ray cluster by default. You can configure your Ray cluster and Serve application to guard them against cluster-level failures as well.
```

Comment:
Suggestion: Can we briefly list some common examples of within cluster and cluster-level failures here?

Reply:
I ended up removing this intro.